src/Compilers/.editorconfig

```diff
@@ -12,6 +12,15 @@ dotnet_diagnostic.RS0101.severity = none
 # RS0102: Braces must not have blank lines between them
 dotnet_diagnostic.RS0102.severity = none
 
+# IDE0059: Unnecessary assignment to a value
+dotnet_diagnostic.IDE0059.severity = none
+
+dotnet_diagnostic.IDE2000.severity = none
+dotnet_diagnostic.IDE2001.severity = none
+dotnet_diagnostic.IDE2002.severity = none
+dotnet_diagnostic.IDE2003.severity = none
+dotnet_diagnostic.IDE2004.severity = none
```

Comment:
All of these lines can be removed.

* IDE0059 is not set to warning for the compiler folder at the higher level, so the override here is unnecessary
* IDE2000-IDE2004 should only be set for the IDE folders at the higher level, so the overrides here are unnecessary